agent/taskresource/fsxwindowsfileserver/fsxwindowsfileserver_windows_test.go

```diff
@@ -458,4 +460,241 @@ func TestSetRootDirectory(t *testing.T) {
+	asmClientCreator := mock_asm_factory.NewMockClientCreator(ctrl)
+	fsxClientCreator := mock_fsx_factory.NewMockFSxClientCreator(ctrl)
+
+	fv := &FSxWindowsFileServerResource{
+		VolumeConfig:FSxWindowsFileServerVolumeConfig{
+			FileSystemID:                 fileSystemId,
+			RootDirectory:                rootDirectory,
+			AuthConfig:                   FSxWindowsFileServerAuthConfig{
+				CredentialsParameter: "arn:aws:foo:us-east-1:123456789012:parameter/test",
+				Domain: domain,
+			},
+			FSxWindowsFileServerHostPath: hostPath,
+		},
+		knownStatusUnsafe:   resourcestatus.ResourceCreated,
+		desiredStatusUnsafe: resourcestatus.ResourceCreated,
+		taskARN:             taskARN,
+		executionCredentialsID: executionCredentialsID,
+	}
+	fv.Initialize(&taskresource.ResourceFields{
+		ResourceFieldsCommon: &taskresource.ResourceFieldsCommon{
+			SSMClientCreator:   ssmClientCreator,
+			ASMClientCreator:   asmClientCreator,
+			FSxClientCreator:   fsxClientCreator,
+			CredentialsManager: credentialsManager,
+		},
+	}, apitaskstatus.TaskStatusNone, apitaskstatus.TaskRunning)
+
+	creds := credentials.TaskIAMRoleCredentials{
+		ARN: "arn",
+		IAMRoleCredentials: credentials.IAMRoleCredentials{
+			AccessKeyID:     "id",
+			SecretAccessKey: "key",
+		},
+	}
+
+	gomock.InOrder(
+		credentialsManager.EXPECT().GetTaskCredentials(gomock.Any()).Return(creds, true),
+	)
+
+	err := fv.Create()
+	assert.Error(t, err)
+}
+
+func TestCreateSSM(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	credentialsManager := mock_credentials.NewMockManager(ctrl)
+	ssmClientCreator := mock_ssm_factory.NewMockSSMClientCreator(ctrl)
+	asmClientCreator := mock_asm_factory.NewMockClientCreator(ctrl)
+	fsxClientCreator := mock_fsx_factory.NewMockFSxClientCreator(ctrl)
+	mockSSMClient := mock_ssmiface.NewMockSSMClient(ctrl)
+	mockFSxClient := mock_fsxiface.NewMockFSxAPI(ctrl)
+
+	fv := &FSxWindowsFileServerResource{
+		VolumeConfig:FSxWindowsFileServerVolumeConfig{
+			FileSystemID:                 fileSystemId,
+			RootDirectory:                rootDirectory,
+			AuthConfig:                   FSxWindowsFileServerAuthConfig{
+				CredentialsParameter: "arn:aws:ssm:us-west-2:123456789012:parameter/test",
+				Domain: domain,
+			},
+			FSxWindowsFileServerHostPath: hostPath,
+		},
+		knownStatusUnsafe:   resourcestatus.ResourceCreated,
+		desiredStatusUnsafe: resourcestatus.ResourceCreated,
+		taskARN:             taskARN,
+		executionCredentialsID: executionCredentialsID,
+	}
+	fv.Initialize(&taskresource.ResourceFields{
+		ResourceFieldsCommon: &taskresource.ResourceFieldsCommon{
+			SSMClientCreator:   ssmClientCreator,
+			ASMClientCreator:   asmClientCreator,
+			FSxClientCreator:   fsxClientCreator,
+			CredentialsManager: credentialsManager,
+		},
+	}, apitaskstatus.TaskStatusNone, apitaskstatus.TaskRunning)
+
+	ssmTestData := "{\n\"username\": \"user\", \n\"password\": \"pass\"\n}"
```

Comment:
I'm not totally clear on what we're using SSM (parameter store?) for.  Is this the AD creds?

Reply:
yes